packages/zendesk-support-adapter/src/filters/field_references.ts

```diff
@@ -0,0 +1,104 @@
+/*
+*                      Copyright 2021 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import { Element } from '@salto-io/adapter-api'
+import { references as referenceUtils } from '@salto-io/adapter-components'
+import { FilterCreator } from '../filter'
+
+const { contextStrategyDefaultLookup, neighborContextGetter } = referenceUtils
+
+const neighborContextFunc = (args: {
+  contextFieldName: string
+  levelsUp?: number
+  contextValueMapper?: referenceUtils.ContextValueMapperFunc
+}): referenceUtils.ContextFunc => neighborContextGetter({
+  ...args,
+  getLookUpName: async ({ ref }) => ref.elemID.name,
+})
+
+const getValueLookupType = (val: string): string | undefined => {
+  const valParts = val.split('_')
+  const lastPart = valParts.pop()
+  if (lastPart === undefined || !['id', 'ids'].includes(lastPart)) {
+    return undefined
+  }
+  return valParts.join('_')
+}
+
+const getLowerCaseSingularLookupType = (val: string): string | undefined => {
+  const lowercaseVal = val.toLowerCase()
+  // for now this simple conversion to singular form seems good enough, but
+  // we may need to improve it later on
+  if (lowercaseVal.endsWith('s')) {
+    return lowercaseVal.slice(0, -1)
+  }
+  return lowercaseVal
+}
+
+export type ReferenceContextStrategyName = referenceUtils.ReferenceContextStrategyName | 'neighborField' | 'neighborType' | 'parentSubject' | 'parentTitle' | 'parentValue'
+export const contextStrategyLookup: Record<
+  ReferenceContextStrategyName, referenceUtils.ContextFunc
+> = {
+  none: contextStrategyDefaultLookup.none,
+  neighborField: neighborContextFunc({ contextFieldName: 'field', contextValueMapper: getValueLookupType }),
+  neighborType: neighborContextFunc({ contextFieldName: 'type', contextValueMapper: getLowerCaseSingularLookupType }),
+  parentSubject: neighborContextFunc({ contextFieldName: 'subject', levelsUp: 1, contextValueMapper: getValueLookupType }),
+  parentTitle: neighborContextFunc({ contextFieldName: 'title', levelsUp: 1, contextValueMapper: getValueLookupType }),
+  parentValue: neighborContextFunc({ contextFieldName: 'value', levelsUp: 2, contextValueMapper: getValueLookupType }),
+}
+
+const fieldNameToTypeMappingDefs: referenceUtils.FieldReferenceDefinition<
+  ReferenceContextStrategyName
+>[] = [
+  { src: { field: 'brand_id' }, serializationStrategy: 'id', target: { type: 'brand' } },
```

Comment:
I prefer the line-breaks way, but it's a matter of taste
```suggestion
{ 
  src: { field: 'brand_id' },
  serializationStrategy: 'id',
  target: { type: 'brand' },
},
```